java/src/main/java/ai/onnxruntime/OnnxRuntime.java

```diff
@@ -139,7 +143,14 @@ static synchronized void init() throws IOException {
       load(ONNXRUNTIME_LIBRARY_NAME);
       load(ONNXRUNTIME_JNI_LIBRARY_NAME);
       ortApiHandle = initialiseAPIBase(ORT_API_VERSION_11);
+      if (ortApiHandle == 0L) {
+        throw new IllegalStateException("Failed to load native library");
+      }
       providers = initialiseProviders(ortApiHandle);
+      version = initialiseVersion();
+      if (version == null) {
+        throw new IllegalStateException("Failed to load version string from native library");
+      }
```

Comment:
This should not generally fail, it does not even return an error code. However, I feel that running this and caching it and risking potential failure, even though most of the people do not need it, is not practical. We only had 1 GH issue that requested it.